src/game/components/inventory.rs

```diff
@@ -246,10 +246,16 @@ impl Inventory {
             self.money = self.money - money;
             Ok(money)
         } else {
-            Err(())
+            let before = self.money;
+            self.money = Money(0);
+            Ok(before)
         }
     }
 
+    pub fn get_money(&self) -> Money {
```

Comment:
I've not really been using getters throughout the code, so just use inventory.money instead of .get_money().

The only place we have simple getters is in AbilityValues and thats because some fields can have additional status effect values added, but some others don't.

Other getters like the get_page below exist because they're not just simple accessing of a field.